### ISSUE TITLE
[Help Wanted] Getting "Cannot GET /" and "Unknown Error" after setting up a self-hosted server

### ISSUE THREAD

**Issue by adamcfe** [standardnotes/forum#1459]
Hi folks, I am struggling with setting up a self-hosted server. I guess I may be missing something quite basic, so I'd be grateful if more experienced eyes could take a look into it with me. Many thanks!

**Describe the issue**
I followed the [official instructions](https://docs.standardnotes.com/self-hosting/docker/) for self-hosting with Docker. I got an "OK" after running `curl http://localhost:3000/healthcheck` but "Cannot GET /" after using `curl http://localhost:3000/` (with or without the slash at the end, if that matters). Desktop client and Android mobile client return "Unknown Error" when trying to register a new account. 

**To Reproduce**
1. Follow the instructions (install everything as root). 
2. Use `curl http://localhost:3000/healthcheck` and get OK. 
3. Use `curl http://localhost:3000/` or `curl http://localhost:3000`.
4. See the following:

> \<!DOCTYPE html>
> \<html lang="en">
> \<head>
> \<meta charset="utf-8">
> \<title>Error</title>
> \</head>
> \<body>
> \<pre>Cannot GET /</pre>
> \</body>
> \</html>

...or just "Cannot GET /" in the browser. 

**Expected behavior**
Devices should be able to register new accounts. As to what I should see instead of "Cannot GET /", I do not know. 

**Desktop (please complete the following information):**
 - OS: Debian 10

Logs (level set to `all` in .env and api-gateway.env, and to `"all"` in auth.env) are available [here.](https://github.com/standardnotes/forum/files/7554489/2021_11_17.-.SN.server.logs.txt)

**Comment by moughxyz** [comment]
This is the correct behavior in the browser. You should plug http://localhost:3000 in the app as custom sync server.

Open Dev Tools > Network, and see what error you get when you try to register.

**Comment by adamcfe** [comment]
I have a headless server so cannot directly enter http://localhost:3000 in Standard Notes settings (I can only do so on a client on a different machine). I have an apache2 reverse proxy configured which routes all traffic through an https address to http://localhost:3000. I include [the configuration](https://github.com/standardnotes/forum/files/7556678/2021_11_17.-.Virtual.hosts.configuration.txt) thereof, just to be sure. 

I am not quite sure what you propose me to do. I assume you meant trying to register the account and observe the Network tab in Firefox. I was not able to see much of it as the server **miraculously started working**. 

I was reading and re-reading the .env files and did a routine restart of the service with `./server.sh stop && ./server.sh start` for no real reason. I am sure I did it before without introducing any changes. I then tried to register an account through app.standardnotes.com and it worked, and it was also possible to login from desktop and register another account from an Android client. 

Well, I guess this is the most ambivalent turn it could have taken, as I am still none the wiser as to what was wrong, but at least it works. 

Thank you so much for your time. You are the best. 

PS: Do I understand correctly that it would not be possible to use the same subdomain I have for the server also for the web app? I think about what to do with the address, as it now sits idle and will probably show "Cannot GET /" until the heat death of the universe. I would much rather use it for the web app's interface or some other worthy cause.
